performance-ptbr.md

```diff
@@ -181,6 +181,83 @@ Idéias para melhorar sua estrutura de dados:
+
+
+Vários algoritmos para funções trigonométricas usam tabelas de pesquisa como um
+ponto de partida para realizar um cálculo.
+
+
+Se o seu programa usa muita memória, também é possível seguir  outro caminho.
+Reduza o uso de espaço em troca do aumento da computação. Em vez de armazenar
+coisas, calcule-as sempre. Você também pode compactar os dados na memória
+e descompactar rapidamente quando precisar.
+
+Se os dados que você está processando estiverem em disco, em vez de carregar tudo na memória
+RAM, você pode criar um índice para as peças necessárias e mantê-las
+memória ou pré-processe o arquivo em pequenos pedaços viáveis.
+
+[Small Memory Software](http://smallmemory.com/book.html)  é um livro disponível online que cobre técnicas utilizadas para o reduzir o espaço usado por seus programas.
+Embora tenha sido originalmente escrito para desenvolvedores de software embarcado, as idéias são
+aplicáveis a programas que rodam em hardware moderno que lidam com grandes quantidades de dados.
+
+* Reorganize seus dados
+
+  Elimine o preenchimento da estrutura. Remova campos extras. Use um tipo de dados menor.
+
+* Mude para uma estrutura de dados mais lenta
+
+  Estruturas de dados mais simples freqüentemente têm requisitos de memória mais baixos. Por exemplo, mudar de uma estrutura de árvore pesada com ponteiro para usar busca linear e slice em arrays.
+  
+* Formato de compactação personalizado para seus dados
+
+  Algoritmos de compressão dependem muito do que está sendo compactado. É
+  melhor escolher um que combine com seus dados. Se você tiver [] byte, algo
+  como snappy, gzip, lz4, se comporta bem. Para dados de ponto flutuante, existe go-tsz
+  para séries temporais e fpc para dados científicos. Muita pesquisa foi feita
+  compactar números inteiros, geralmente para recuperação de informações em motores de pesquisa. Exemplos incluem codificação delta e varints para esquemas mais complexos envolvendo Huffman códificado com diferenças de OU-exclusivo. Você também pode criar
+  formatos de compactação otimizados para seus tipos exatos de dados.
+
+
+Você precisa inspecionar os dados ou eles podem permanecer compactados? Você precisa de acesso aleatório ou apenas streaming? 
+Se você precisar acessar entradas individuais, mas não quer descomprimir a coisa toda, você pode compactar os dados em blocos menores e manter um índice indicando o intervalo de entradas em cada bloco. O acesso a uma única entrada só precisa verificar o índice e descompactar o bloco de dados menor.
+
+Se seus dados não estão apenas sendo processados, mas também serão gravados em disco, que tal  migração de dados ou adição / remoção de campos. Agora você estará lidando com os [] byte em sua forma crua, em vez de bons tipos estruturados de Go, portanto, você precisará não ser seguro e considerar as opções de serialização.
+
```

Comment:
Maybe you need another blank line, as the following line should get the indentation reset.